ui.apps/src/main/content/jcr_root/apps/core/cif/clientlibs/storefront-events/js/collector.js

```diff
@@ -31,4 +31,17 @@ const processProductStorefrontData = () => {
     }
 };
 
+const processSearchInputStorefrontData = () => {
+    const searchInputCtxElement = document.querySelector('[data-cif-search-input-context]');
+    if (searchInputCtxElement) {
+        try {
+            const searchInputCtx = JSON.parse(searchInputCtxElement.dataset.cifSearchInputContext);
+            mse.context.setSearchInput(searchInputCtx);
```

Comment:
Are you sure this fits the schema? IIUC you pass a [SearchInputUnit](https://github.com/adobe/magento-storefront-events-sdk/blob/main/src/types/schemas/searchInput.ts#L10-L19) here, but instead you should be passing a [SearchInput](https://github.com/adobe/magento-storefront-events-sdk/blob/main/src/types/schemas/searchInput.ts#L6-L8). Like I don't see the `units` array anywhere :)

Reply:
yup, I missed it. I also missed a couple of fields from SearchInput. 
Fixed it